unit-tests/run-unit-tests.py

```diff
@@ -132,11 +132,14 @@ def subprocess_run(cmd, stdout = None):
         if stdout  and  stdout != subprocess.PIPE:
             handle = open( stdout, "w" )
             stdout = handle
+        start_time = time.time()
         rv = subprocess.run( cmd,
                              stdout = stdout,
                              stderr = subprocess.STDOUT,
                              universal_newlines = True,
+                             timeout=200,

```

Comment:
What's the unit here?
Why 200?

Reply:
Seconds? 3m20s?

Reply:
Its in seconds, you said to give about 3 minutes, 200 seemed like a nice round number

Reply:
So please note the units.
Preferrably, add a parameter `timeout` and note the units in its description/name

Reply:
Please also add some short doc to the function that states its common return/exit scenarios

Reply:
by parameter do you mean just a global variable or an option like --regex?

Reply:
It's a generic function -- add it as a function argument?